backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/CassandraAsyncExecutor.java

```diff
@@ -42,9 +46,9 @@ public CassandraAsyncExecutor(Session session) {
     }
 
     public Mono<ResultSet> execute(Statement statement) {
-        return Mono.defer(() -> Mono.fromFuture(FutureConverter
+        return retryOnError(Mono.defer(() -> Mono.fromFuture(FutureConverter
```

Comment:
how do you know you can retry? Some operations are idempotent, but some other are not, no? For example increasing quota count.

Reply:
Sure, but you do not know that it has been run or not.

Reply:
Yes, that's my point. Retry should not be for all operations, just for operations that you know they can be retried.